lstchain/calib/camera/r0.py

```diff
@@ -0,0 +1,288 @@
+        event.r0.tel[self.telid].waveform.astype(np.uint16)
+
+    def time_lapse_corr(self, event):
+        """
+        Perform time lapse corrections.
+        Parameters
+        ----------
+        event : `ctapipe` event-container
+        """
+        expected_pixel_id = event.lst.tel[0].svc.pixel_ids
+        local_clock_list = event.lst.tel[0].evt.local_clock_counter
+        n_modules = event.lst.tel[0].svc.num_modules
+        for nr_clus in range(0, n_modules):
+            self.first_cap_time_lapse_array[nr_clus, :, :] = self._get_first_capacitor(event, nr_clus)
+
+        do_time_lapse_corr(event.r0.tel[0].waveform, expected_pixel_id, local_clock_list,
+                           self.first_cap_time_lapse_array, self.last_reading_time_array, n_modules)
+        event.r0.tel[self.telid].waveform.astype(np.uint16)
+
+    def interpolate_spikes(self, event):
+        """
+        Interpolates spike A & B and change the R0 container.
+
+        Parameters
+        ----------
+        event : `ctapipe` event-container
+        """
+        self.first_cap_old_array[:, :, :] = self.first_cap_array_spike[:, :, :]
+        n_modules = event.lst.tel[0].svc.num_modules
+        for nr_module in range(0, n_modules):
+            self.first_cap_array_spike[nr_module, :, :] = self._get_first_capacitor(event, nr_module)
+
+        waveform = event.r0.tel[0].waveform[:, :, :]
+        expected_pixel_id = event.lst.tel[0].svc.pixel_ids
+        wf = waveform.copy()
+        wf = wf.astype('int16')
+        event.r0.tel[0].waveform = self.interpolate_pseudo_pulses(wf,
+                                                                  expected_pixel_id,
+                                                                  self.first_cap_array_spike,
+                                                                  self.first_cap_old_array,
+                                                                  n_modules)
+        event.r0.tel[self.telid].waveform.astype(np.uint16)
+
+    @staticmethod
+    @jit(parallel=True)
+    def interpolate_pseudo_pulses(waveform, expected_pixel_id, fc, fc_old, n_modules):
+        """
+        Interpolate Spike A & B and change the R0 container.
+
+        Parameters
+        ----------
+        waveform : ndarray
+            Waveform stored in a numpy array of shape
+            (n_gain, n_pix, n_samples).
+        expected_pixel_id: ndarray
+            Array stored expected pixel id
+            (n_pix*n_modules).
+        fc : ndarray
+            Value of first capacitor stored in a numpy array of shape
+            (n_clus, n_gain, n_pix).
+        fc_old : ndarray
+            Value of first capacitor from previous event
+            stored in a numpy array of shape
+            (n_clus, n_gain, n_pix).
+        n_modules : int
+            Number of modules
+        """
+        roisize = 40
+        size4drs = 4096
+        n_gain = 2
+        n_pix = 7
+        for nr_clus in prange(0, n_modules):
```

Comment:
Same as comment below about the loops, although here it seems to be more difficult to simplify...

Reply:
@rlopezcoto Thank you,
I will improve code according comments.